documentation/service_onboarding.md

```diff
@@ -0,0 +1,164 @@
+
+### Constructors
+All clients should be able to be initialized directly from the user and should begin with `New`. For example to define a constructor for a new client for the Tables service we start with defining the struct `TableServiceClient`:
+```golang
+// A TableServiceClient represents a client to the table service. It can be used to query the available tables, add/remove tables, and various other service level operations.
+type TableServiceClient struct {
+	client  *tableClient
+	service *serviceClient
+	cred    SharedKeyCredential
+}
+
+```
+Note that there are no exported fields on the `TableServiceClient` struct, and as a rule of thumb, generated clients and credentials should be private.
+
+Constructors for clients are separate methods that are not associated with the struct. The constructor for the TableServiceClient is as follow:
+```golang
+// NewTableServiceClient creates a TableServiceClient struct using the specified serviceURL, credential, and options.
+func NewTableServiceClient(serviceURL string, credential azcore.Credential, options *TableServiceClientOptions) (*TableServiceClient, error) {
+	conOptions := options.getConnectionOptions()
+	if isCosmosEndpoint(serviceURL) {
+		conOptions.PerCallPolicies = []azcore.Policy{CosmosPatchTransformPolicy{}}
+	}
+	con := newConnection(serviceURL, cred, conOptions)
+	c, err := cred.(*SharedKeyCredential)
+	return &TableServiceClient{client: &tableClient{con}, service: &serviceClient{con}, cred: *c}, err
+}
+```
+In `Go`, the method parameters are enclosed with parenthesis immediately following the method name with the parameter name preceding the parameter type. The return arguments follow the parameters. If a method has more than one return parameter the types of the parameter must be enclosed in parenthesis. Note the `*` before a type indicates a pointer to that type. All methods that create a new client or interact with the service should return an `error` type as the last argument.
+
+This client takes three parameters, the first is the service URL for the specific account. The second is an [`interface`](https://gobyexample.com/interfaces) which is a specific struct that has definitions for a certain set of methods. In the case of `azcore.Credential` the `AuthenticationPolicy(options AuthenticationPolicyOptions) Policy` method must be defined to be a valid interface. The final argument to methods that create clients or interact with the service should be a pointer to an `Options` parameter. Making this final parameter a pointer allows the customer to pass in `nil` if there are no specific options they want to change. The `Options` type should have a name that is intuitive to what the customer is trying to do, in this case `TableClientOptions`.
+
+### Defining Methods
+Defining a method follows the format:
+```golang
+func (m *<MyStruct>) MethodName(param1 param1Type, param2 param2Type) (ReturnType, ReturnType2) {
```

Comment:
Since you've showed a real constructor above you can also use it here (ie, instead of <MyStruct> just use TableServiceClient). If you're reading it linearly it's helpful, and even if you just dive right into this it's illustrative, I think.